src/merkle/partial_mt/tests.rs

```diff
@@ -0,0 +1,221 @@
+use crate::hash::rpo::RpoDigest;
+
+use super::{
+    super::{int_to_node, NodeIndex},
+    InnerNodeInfo, MerkleError, PartialMerkleTree, Rpo256, Vec, Word,
+};
+
+// TEST DATA
+// ================================================================================================
+
+const ROOT_NODE: NodeIndex = NodeIndex::new_unchecked(0, 0);
+
+const NODE10: NodeIndex = NodeIndex::new_unchecked(1, 0);
+const NODE11: NodeIndex = NodeIndex::new_unchecked(1, 1);
+
+const NODE20: NodeIndex = NodeIndex::new_unchecked(2, 0);
+const NODE21: NodeIndex = NodeIndex::new_unchecked(2, 1);
+const NODE22: NodeIndex = NodeIndex::new_unchecked(2, 2);
+const NODE23: NodeIndex = NodeIndex::new_unchecked(2, 3);
+
+const NODE32: NodeIndex = NodeIndex::new_unchecked(3, 2);
+const NODE33: NodeIndex = NodeIndex::new_unchecked(3, 3);
+
+// TESTS
+// ================================================================================================
+
+// with_paths CONSTRUCTOR TESTS
+// ------------------------------------------------------------------------------------------------
+
+#[test]
+fn get_root() {
+    let leaf0 = int_to_node(0);
+    let leaf1 = int_to_node(1);
+    let leaf2 = int_to_node(2);
+    let leaf3 = int_to_node(3);
+
+    let parent0 = calculate_parent_hash(leaf0, 0, leaf1);
+    let parent1 = calculate_parent_hash(leaf2, 2, leaf3);
+
+    let root_exp = calculate_parent_hash(parent0, 0, parent1);
+
+    let set = super::PartialMerkleTree::with_paths([(NODE20, leaf0, vec![leaf1, parent1].into())])
+        .unwrap();
+
+    assert_eq!(set.root(), root_exp);
+}
+
+#[test]
+fn add_and_get_paths() {
+    let value32 = int_to_node(32).into();
+    let value33 = int_to_node(33).into();
+    let value20 = int_to_node(20).into();
+    let value22 = int_to_node(22).into();
+    let value23 = int_to_node(23).into();
+
+    let value21 = Rpo256::merge(&[value32, value33]);
+    let value10 = Rpo256::merge(&[value20, value21]);
+    let value11 = Rpo256::merge(&[value22, value23]);
+
+    let path_33 = vec![*value32, *value20, *value11];
+
+    let path_22 = vec![*value23, *value10];
```

Comment:
Instead of doing this by hand, I wonder if it would be better to test against `MerkleStore` as it also has the ability to handle paths of different depths. (this is applicable to most tests in this file).

Reply:
I don't think that it will be easier. If I correctly understood how to create a `MerkleStore` instance, first we need to create paths, which then will be used to create a `MerkleStore`, so we need to calculate most of these nodes anyway. `MerkleStore` can help to check correctness of `PartialMerkleTree` functions output, but that seems to be useless since we already calculated all necessary nodes and values to create `MerkleStore` and `PartialMerkleTree`. Or maybe I just don't understand how we can use `MerkleStore` here.

Reply:
I guess it would depend on how we do it. We could create a `MerkleTree` (let's say of depth 3 or 4) and then create a `MerkleStore` from this Merkle tree. Then, we could get Merkle paths of various depths from this `MerkleStore` and create `PartialMerkleTree` from it. Then we could update the data in the Merkle store and make sure that similar updates to the Partial Merkle tree would give the same results.

Basically, I'm trying to avoid creating Merkle paths manually because we already have (tested) code which does that.